### PR TITLE
skip process isolation check for transmit and process

### LINKED ISSUE #1085: while running via the transmit-worker-process, ansible-runner should only check process isolation exec for worker
When trying to run job via `transmit-worker-process` while using `process_isolation_executable` the kwargs pass into 
`transmit` will contain the `process_isolation=true` this cause `check_isolation_executable_installed` to run during all 3 of the phases (transmit, worker, process) 

https://github.com/ansible/ansible-runner/blob/a349921448f84dc507abbd87aca163291f2c3bdc/ansible_runner/interface.py#L86-L90

In order for the `transmit-worker-process`  chain to use `process_isolation_executable` the argument must be passed to `transmit` and have `transmit` pass the argument down the chain. Therefore we can not just simply remove the argument.

In AWX `ansible-runner transmit` runs on host that `dispatcher` runs while `ansible-runner worker` runs on host that the `receptor` runs

It is possible for the host for `dispatcher` does not have the `process_isolation_executable` (since the job is not actually run on the host) and only the host for `receptor` have the `process_isolation_executable`

https://github.com/ansible/awx/blob/4dc956c76f31accc3f57599d5e6df1735ffa2c16/awx/main/tasks/receptor.py#L391

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
https://github.com/ansible/ansible-runner/issues/1085

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ rpm-build/
 .tox/
 .venv
 venv/
+.vscode/
```
Comment (shanemcd) [inline .gitignore]: Can you split this out into a different commit?

Comment (Shrews) [reply .gitignore]: This should be in the user's `$HOME/.gitignore_global` file, IMO, not a per-repo ignore.

File: ansible_runner/interface.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def init_runner(**kwargs):
         if logfile:
             output.set_logfile(logfile)
 
-    if kwargs.get("process_isolation", False):
+    if kwargs.get("process_isolation", False) and kwargs.get("streamer") not in ('transmit', 'process'):
```
Comment (matburt) [inline ansible_runner/interface.py]: I have this in my PR also: https://github.com/ansible/ansible-runner/pull/1079/files#diff-179ccecedba1c6ccf73b8c02f0321b252c018226e41ce82887a93ace762d2fbbR55

This line causes an astonishing amount of problems when interfacing with Runner from the CLI

**Comment by matburt** [comment]
Fix that vscode comment and then I think we should merge this asap.

File: ansible_runner/interface.py (hunk 2 above)
Comment (matburt) [reply ansible_runner/interface.py]: I think the reason it does this has to do with the rest of my PR, which is that the CLI entrypoint clashes with the module level entrypoint when managing process isolation paths.

My work on #1079  focuses on realigning those two paths.

**Comment by TheRealHaoLiu** [comment]
`.gitignore` change amended as per suggestion

**Comment by TheRealHaoLiu** [comment]
code change after peer programing with @jbradberry moved the location of the check

**Review by jbradberry — APPROVED** [review]

**Comment by Shrews** [comment]
Manually tested this to verify that `worker` will bail if the executable is not found. I'd like to have a test for this before merging it. Thanks!

**Comment by TheRealHaoLiu** [comment]
> Manually tested this to verify that `worker` will bail if the executable is not found. I'd like to have a test for this before merging it. Thanks!

testcase added

File: test/integration/test_transmit_worker_process.py [hunk 3]
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
...
+            _output=outgoing_buffer,
+            **job_kwargs,
+        )
+
+        # valide process_isolation kwargs are passed to transmitter
+        assert transmitter.kwargs['process_isolation'] == job_kwargs['process_isolation']
+        assert transmitter.kwargs['process_isolation_executable'] == job_kwargs['process_isolation_executable']
+
+        # validate that transmit did not fail due to missing process isolation executable
+        assert transmitter.rc in (None, 0)
```
Comment (Shrews) [inline test/integration/test_transmit_worker_process.py:196]: Is `None` really a valid value here?

File: test/integration/test_transmit_worker_process.py [hunk 4]
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
...
+        assert b'kwargs' in sent
+
+        # validate kwargs in buffer contain correct process_isolation and process_isolation_executable
+        for line in sent.decode('utf-8').split('\n'):
+            if "kwargs" in line:
+                kwargs = json.loads(line).get("kwargs", {})
+                assert kwargs['process_isolation'] is True
+                assert kwargs['process_isolation_executable'] == process_isolation_executable
+                break
+
```
Comment (Shrews) [inline test/integration/test_transmit_worker_process.py:213]: Similar to the `transmit` check above, you should check that `worker` will fail if the executable is missing.

**Review by Shrews — CHANGES_REQUESTED** [review]

File: test/integration/test_transmit_worker_process.py [hunk 5]
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
...
         self.check_artifacts(str(process_dir), job_type)
 
+    @pytest.mark.parametrize("process_isolation_executable", ['exist', 'does_not_exist'])
+    def test_process_isolation_executable_transmit(self, tmp_path, process_isolation_executable, mocker):
+        """
+        Case transmit should not fail if process isolation executable does not exist
+        """
+        def mock_check_isolation_executable_installed(executable):
+            if executable == "exist":
+                return True
```
Comment (Shrews) [inline test/integration/test_transmit_worker_process.py]:
```suggestion
                return True
            return False
```

File: test/integration/test_transmit_worker_process.py (hunk 3 above)
Comment (TheRealHaoLiu) [reply test/integration/test_transmit_worker_process.py:196]: https://github.com/ansible/ansible-runner/blob/devel/test/integration/test_transmit_worker_process.py#L74-L75

**Comment by TheRealHaoLiu** [comment]
testcase commit amended base on @Shrews's feedback

**Review by Shrews — APPROVED** [review]
Thanks for the PR! If you want this in any of the previously released 2.x versions, you'll need to supply a backport to the appropriate release branches. We follow the same basic backport process as Ansible (https://docs.ansible.com/ansible/latest/community/development_process.html#backport-process).

**Comment by TheRealHaoLiu** [comment]
> Thanks for the PR! If you want this in any of the previously released 2.x versions, you'll need to supply a backport to the appropriate release branches. We follow the same basic backport process as Ansible (https://docs.ansible.com/ansible/latest/community/development_process.html#backport-process).

no need for backport thanks

**Comment by TheRealHaoLiu** [comment]
backport to 2.2 https://github.com/ansible/ansible-runner/pull/1094
